### DISCUSSION TITLE
Workflow for layered requirements: Is it possible to define 2 separate sets of dependencies?

### DISCUSSION THREAD

**Discussion by AdamJel** [python-poetry/poetry#5544] [Ideas]
Consider a situation, when a project is going through an overhaul and in the meantime, it serves 2 interfaces at once - the old one, not yet completely removed, and the new one, not yet completely implemented. For this reason, it is just one project, one repository. But there are already two sets of dependencies, which differ substantially. And I would like to build two container images - one for each set. That is an image for old interface with old dependencies and second image with new interface and new dependencies.

Is it possible to achieve this with poetry?

I know about [dependencies group within toml file](https://python-poetry.org/docs/master/managing-dependencies/#dependency-groups). But this does not solve the problem, since they are rather labels, then true groups (the dependency graph is resolved across groups).

A possible solution (used in [pip-tools](https://pip-tools.readthedocs.io/en/latest/#workflow-for-layered-requirements) would be to use different toml files -> to have 2 toml files `pyproject-a.toml` and `pyproject-b.toml` and two separate lock files as a result. But AFAIK this is not possible with poetry, or is it?

Is there any other way/workaround how to achieve this? Except for splitting the repository into 2 independent projects, which we don't want to do right now.

Another possible way to do this (a possible feature request) would be, to allow resolving the graph within export command for specified group. E.g.

```bash
export --group "new-interface" --output "requirements-new-interface.txt"
```

What do you think @abn and all?

**Comment by alexlatchford** [comment]
+1 to this idea, we currently utilize `pip-tools` in these cases but would love it if `poetry` had this feature as we use `poetry` everywhere else!
